runtime/vm/jfr.cpp

```diff
@@ -602,15 +662,56 @@ initializeEventFields(J9VMThread *currentThread, J9JFREvent *event, UDATA eventT
 	event->vmThread = currentThread;
 }
 
+// TODO: add thread state parameter
 void
 jfrExecutionSample(J9VMThread *currentThread, J9VMThread *sampleThread)
 {
+#if defined(DEBUG)
+	PORT_ACCESS_FROM_VMC(currentThread);
+	j9tty_printf(PORTLIB, "\n!!! execution sample %p %p\n", currentThread, sampleThread);
+#endif /* defined(DEBUG) */
+
 	J9JFRExecutionSample *jfrEvent = (J9JFRExecutionSample*)reserveBufferWithStackTrace(currentThread, sampleThread, J9JFR_EVENT_TYPE_EXECUTION_SAMPLE, sizeof(*jfrEvent));
 	if (NULL != jfrEvent) {
 		jfrEvent->threadState = J9JFR_THREAD_STATE_RUNNING;
 	}
 }
 
+static int J9THREAD_PROC
+jfrSamplingThreadProc(void *entryArg)
+{
+	J9JavaVM *vm = (J9JavaVM*)entryArg;
+	J9VMThread *currentThread = NULL;
+
+	if (JNI_OK == attachSystemDaemonThread(vm, &currentThread, "JFR sampler")) {
+		omrthread_monitor_enter(vm->jfrSamplerMutex);
+		vm->jfrSamplerState = J9JFR_SAMPLER_STATE_RUNNING;
+		omrthread_monitor_notify_all(vm->jfrSamplerMutex);
+		while (J9JFR_SAMPLER_STATE_STOP != vm->jfrSamplerState) {
+			omrthread_monitor_exit(vm->jfrSamplerMutex);
+			internalEnterVMFromJNI(currentThread);
+			acquireExclusiveVMAccess(currentThread);
+			J9VMThread *walkThread = J9_LINKED_LIST_START_DO(vm->mainThread);
+			while (NULL != walkThread) {
+				jfrExecutionSample(currentThread, walkThread);
```

Comment:
im seeing a lot of empty stack traces, perhaps we should wrap this with

```
				if (VM_VMHelpers::threadCanRunJavaCode(walkThread)) {
					jfrExecutionSample(currentThread, walkThread);
				}
```

Reply:
Done.